### PR TITLE
Support zstd as compression type for file references

### PR DESCRIPTION
**Pull request by hmusum**

### CODE REVIEW

File: filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReferenceCompressor.java [hunk 1]
```diff
@@ -118,32 +119,34 @@ private static void writeFileToTar(ArchiveOutputStream taos, File baseDir, File
...
-                    default -> throw new RuntimeException("Unknown compression type " + compressionType);
+                    case gzip -> new GZIPOutputStream(out);
+                    case lz4 -> new LZ4BlockOutputStream(out);
+                    case zstd -> new ZstdOutputStream(out);
                 };
             case file:
-                return new FileOutputStream(outputFile);
+                return out;
             default:
                 throw new RuntimeException("Unknown file reference type " + type);
```
Comment (baldersheim) [inline filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReferenceCompressor.java:134]: Need to close out stream here.

File: filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReferenceCompressor.java [hunk 2]
```diff
@@ -118,32 +119,34 @@ private static void writeFileToTar(ArchiveOutputStream taos, File baseDir, File
...
-                    default -> throw new RuntimeException("Unknown compression type " + compressionType);
+                    case gzip -> new GZIPInputStream(in);
+                    case lz4 -> new LZ4BlockInputStream(in);
+                    case zstd -> new ZstdInputStream(in);
                 };
             case file:
-                return new FileInputStream(inputFile);
+                return in;
             default:
                 throw new RuntimeException("Unknown file reference type " + type);
```
Comment (baldersheim) [inline filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReferenceCompressor.java:151]: Need to close in stream here.

**Review by baldersheim — CHANGES_REQUESTED** [review]
Close on bad type

File: filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReferenceCompressor.java (hunk 1 above)
Comment (hmusum) [reply filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReferenceCompressor.java:135]: fixed

File: filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReferenceCompressor.java (hunk 2 above)
Comment (hmusum) [reply filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReferenceCompressor.java:153]: fixed

**Review by baldersheim — APPROVED** [review]

**Pull request merged**